### PR TITLE
fix(runtime-params-estimator): disable rocksdb thread in it

### PR DESCRIPTION
**Pull request by ailisp**
Thanks to an external contributor, we're able to use all export apis to disable rocksdb threads in runtime param estimator

Test Plan
---------
Observe only one thread during run runtime params estimator
image

### CODE REVIEW

**Comment by ailisp** [comment]
Seems parameter estimator get stuck in the middle after turn off all threads, probably due to there's limit of how big the state it can have. will see if i can increase the bar to proceed

File: runtime/runtime-params-estimator/src/cases.rs [hunk 1]
```diff
@@ -150,7 +150,7 @@ pub fn run(mut config: Config) -> RuntimeConfig {
     config.block_sizes = vec![100];
     // Measure the speed of processing empty receipts.
     measure_actions(Metric::Receipt, &mut m, &config, None, vec![], false, false);
-
+    println!("fully finish one measure actions");
```
Comment (willemneal) [inline runtime/runtime-params-estimator/src/cases.rs:153]: Was this just for local debugging?

**Comment by evgenykuzyakov** [comment]
>Why couldn't we only use it for parameter estimator, i.e, putting this change behind some feature flag?

Actually, I agree with @bowenwang1996 on this. But it also can be useful not only for param estimator, so some generic feature name will be good

File: chain/chain/Cargo.toml [hunk 2]
```diff
@@ -10,7 +10,7 @@ log = "0.4"
 failure = "0.1"
 failure_derive = "0.1"
 lazy_static = "1.4"
-rocksdb = "0.14"
+rocksdb = { git = "https://github.com/nearprotocol/rust-rocksdb", branch="disable-thread" }
```
Comment (bowenwang1996) [inline chain/chain/Cargo.toml:13]: why?

Comment (ailisp) [reply chain/chain/Cargo.toml:13]: It use https://github.com/facebook/rocksdb/pull/7191 (merged, but rust-rocksdb haven't bump version) and https://github.com/rust-rocksdb/rust-rocksdb/pull/448 (approved, but need to wait next time rust-rocksdb bump version). Our fork is equal to https://github.com/rust-rocksdb/rust-rocksdb/pull/448

File: core/store/src/db.rs [hunk 3]
```diff
@@ -363,8 +367,22 @@ fn rocksdb_options() -> Options {
     opts.set_bytes_per_sync(1048576);
     opts.set_write_buffer_size(1024 * 1024 * 512 / 2);
     opts.set_max_bytes_for_level_base(1024 * 1024 * 512 / 2);
-    opts.increase_parallelism(cmp::max(1, num_cpus::get() as i32 / 2));
-    opts.set_max_total_wal_size(1 * 1024 * 1024 * 1024);
+    #[cfg(not(feature = "single_thread_rocksdb"))]
+    {
+        println!("Use background threads in rocksdb");
```
Comment (frol) [inline core/store/src/db.rs:372]: Should we remove this log message or convert it into a proper tracing::info/debug/trace?

File: core/store/src/db.rs [hunk 4]
```diff
@@ -419,12 +437,32 @@ impl RocksDB {
...
         let db = DB::open_cf_descriptors(&options, path, cf_descriptors)?;
+        #[cfg(feature = "single_thread_rocksdb")]
+        {
+            // These have to be set after open db
+            let mut env = Env::default().unwrap();
+            env.set_bottom_priority_background_threads(0);
+            env.set_high_priority_background_threads(0);
+            env.set_low_priority_background_threads(0);
+            env.set_background_threads(0);
+            println!("Disabled all background threads in rocksdb");
```
Comment (frol) [inline core/store/src/db.rs:448]: ^

Comment (ailisp) [reply core/store/src/db.rs:448]: I prefer to keep this one to exolicitly know this happens on test. The above one goes into neard, probably should removed

File: core/store/src/db.rs (hunk 3 above)
Comment (ailisp) [reply core/store/src/db.rs:372]: Will convert to debug

Comment (ailisp) [reply core/store/src/db.rs:372]: Seems we didn't log anything in db config module in neard, so I'll remove it to be consistent

**Pull request merged**